stl/inc/memory

```diff
@@ -2351,6 +2369,25 @@ void _Uninitialized_value_construct_multidimensional_n(_Ty* const _Out, const si
     }
 }
 
+template <class _Ty>
+void _Uninitialized_default_construct_multidimensional_n(_Ty* const _Out, const size_t _Size) {
+    if constexpr (!is_trivially_default_constructible_v<_Ty>) {

```

Comment:
To be absolutely sure, even for the `meow_overwrite` we do not want to default initialize here?

Reply:
That's the whole point of `*_for_overwrite`

Reply:
For better or worse, there are many places in the STL where we assume that we can omit default-initialization of objects that are trivially default constructible. I'm not sure the resulting behavior is portably defined, but it has been working as intended on our implementation. We may need to make changes if and when compilers complain about this trick in `constexpr`, but that hasn't happened yet and in any case would need for more widespread changes than just this PR.